### PR TITLE
Support event hub

### PR DESCRIPTION
**Pull request by johnwatson484**

### CODE REVIEW

File: app/event/send-payment-request-invalid-events.js [hunk 1]
```diff
@@ -28,4 +40,23 @@ const sendPaymentRequestInvalidEvents = async (paymentRequests) => {
   }
 }
 
+const sendV2PaymentRequestInvalidEvents = async (paymentRequests) => {
+  if (paymentRequests?.length) {
+    const events = paymentRequests.map(createEvent)
```
Comment (simonnineteen90) [inline app/event/send-payment-request-invalid-events.js:45]: Do we need any validation on the object as on line 23/23 above?

Comment (johnwatson484) [reply app/event/send-payment-request-invalid-events.js:45]: The original validation isn't really adding any value.  It's just checking that the value is an object which isn't really helpful.

With V2 events, the message is validated within the npm package so we don't need to duplicate that in each app.

File: test/unit/event/send-batch-error-event.test.js [hunk 2]
```diff
@@ -13,39 +42,103 @@ const error = {
...
+    expect(mockPublishEvent).toHaveBeenCalled()
+  })
+
+  test('should not send V2 events when v2 events disabled ', async () => {
+    processingConfig.useV2Events = false
+    await sendBatchErrorEvent(filename, error)
+    expect(mockPublishEvent).not.toHaveBeenCalled()
+  })
+
+  test('should send event to V2 topic', async () => {
```
Comment (simonnineteen90) [inline test/unit/event/send-batch-error-event.test.js:119]: do we need an equivalent test for sending to the V1 topic? Only ask as i tried to implement that as a test and couldnt get it to work and wondered why you removed it, is it not possible/easy/worth testing?

File: test/integration/processing/siti-agri/filter-payment-requests.test.js [hunk 3]
```diff
@@ -28,6 +28,7 @@ describe('Filter payment requests', () => {
 
   test('should return mappedPaymentRequest as successfulPaymentRequests when valid payment request and sourceSystem are given', async () => {
     const result = filterPaymentRequest(paymentRequests, sourceSystem)
+    console.log(result)
```
[the remark this answers is not in the document]
Comment (johnwatson484) [reply test/integration/processing/siti-agri/filter-payment-requests.test.js]: No.. my secrets are out!!

Fixed

File: test/unit/event/send-batch-error-event.test.js (hunk 2 above)
Comment (johnwatson484) [reply test/unit/event/send-batch-error-event.test.js:124]: My mistake, took it out to sort the other tests and forgot to re-add it.  Restored now.

**Review by simonnineteen90 — APPROVED** [review]

**Pull request merged**